docs/service-bundle.md

```diff
@@ -0,0 +1,225 @@
+# Service Bundle Specification
+
+A Service Bundle is a container image that the broker can use to manage
+deployment of a service in a cluster. It adheres to the properties and
+behaviors described below.
+
+## Label
+
+Every service bundle container image has a label ‘com.redhat.apb.spec’ that
+contains a base64 encoded string. This label holds [important
+metadata](https://github.com/ansibleplaybookbundle/ansible-playbook-bundle/blob/master/docs/developers.md#apb-spec-file)
+related to the service bundle. Example:
+
+```
+version: 1.0        # The version of the Service Bundle Spec being conformed to
+name: example-apb   # The name of the Service Bundle
+description:        # A short description of the Service Bundle
+bindable: True      # Whether this Service Bundle can be bound to other services
+async: optional     # Does this Service Bundle support asynchronous provision
+metadata: 
+  documentationUrl: <link to documentation>
+  imageUrl: <link to URL of image>
+  dependencies:
+  - '<registry>/<organization>/<dependency-name-1>'   
+  - '<registry>/<organization>/<dependency-name-2>'
+  displayName: Example App (APB)
+  longDescription: A longer description of what this APB does
+  providerDisplayName: "Red Hat, Inc."
+plans: # An array of plans supported by this Service Bundle (must have at least 1)
+  - name: default
+    description: A short description of what this plan does
+    free: true
+    metadata:
+      displayName: Default
+      longDescription: A longer description of what this plan deploys
+      cost: $0.00
+    Parameters: # Parameters for this plan
+      - name: parameter_one
+        required: true
+        default: foo_string
+        type: string
+        title: Parameter One
+        maxlength: 63
+      - name: parameter_two
+        required: true
+        default: true
+        title: Parameter Two
+        type: boolean
+```
+
+## Input
+
+The container is run with three arguments.
+
+* Action: a single word such as “provision”. The full list is below.
+* --extra-vars: This is ansible-specific and always has exactly the value
+  “--extra-vars”. An Ansible Playbook Bundle uses this and the third argument
```

Comment:
This happens to be ansible specific due to our heritage, but do we want to express arguments in a more generic way and have an apb-base perform a transform before delivering them to the playbook?

Reply:
I agree with making args more generic and having Service Bundle base images do the transform.